src/mermaidChartAuthenticationProvider.ts

```diff
@@ -14,80 +13,38 @@ import {
   window,
 } from "vscode";
 import { v4 as uuid } from "uuid";
-import { getEncodedSHA256Hash, PromiseAdapter, promiseFromEvent } from "./util";
-import defaultAxios from "axios";
-
-const test = true;
-export const AUTH_TYPE = `mermaidchart`;
-const AUTH_NAME = `MermaidChart`;
-const CLIENT_ID = `469e30a6-2602-4022-aff8-2ab36842dc57`;
-const SESSIONS_SECRET_KEY = `${AUTH_TYPE}.sessions`;
-const MERMAIDCHART_BASEURL = test
-  ? "http://127.0.0.1:5174"
-  : "https://www.mermaidchart.com";
-
-const axios = defaultAxios.create({
-  baseURL: MERMAIDCHART_BASEURL,
-});
-
-interface OAuthAuthorizationParams {
-  response_type: "code";
-  client_id: string;
-  redirect_uri: string;
-  code_challenge_method: "S256";
-  code_challenge: string;
-  state: string;
-  scope: string;
-}
-
-const URLS = {
-  oauth: {
-    authorize: (params: OAuthAuthorizationParams) =>
-      `/oauth/authorize?${new URLSearchParams(
-        Object.entries(params)
-      ).toString()}`,
-    token: `/oauth/token`,
-  },
-  rest: {
-    users: {
-      self: `/rest-api/users/me`,
-    },
-  },
-} as const;
+import { PromiseAdapter, promiseFromEvent } from "./util";
+import { MermaidChartVSCode } from "./mermaidChartVSCode";
 
 class UriEventHandler extends EventEmitter<Uri> implements UriHandler {
   public handleUri(uri: Uri) {
     this.fire(uri);
   }
 }
 
-interface AuthState {
-  codeVerifier: string;
-}
-
 export class MermaidChartAuthenticationProvider
   implements AuthenticationProvider, Disposable
 {
+  static id = "mermaidchart";
+  static providerName = "MermaidChart";
+  private sessionsKey = `${MermaidChartAuthenticationProvider.id}.sessions`;
   private _sessionChangeEmitter =
     new EventEmitter<AuthenticationProviderAuthenticationSessionsChangeEvent>();
   private _disposable: Disposable;
-  private _pendingStates: Record<string, AuthState> = {};
   private _codeExchangePromises = new Map<
     string,
     { promise: Promise<string>; cancel: EventEmitter<void> }
   >();
   private _uriHandler = new UriEventHandler();
 
-  constructor(private readonly context: ExtensionContext) {
+  constructor(
+    private readonly mcAPI: MermaidChartVSCode,
```

Comment:
Q: For my understanding... So for other plugins, instead of mcAPI: MermaidChartVSCode another api such as "mcAPI: MermaidChartGoogleDocs"?

Reply:
Yes, we'll have to build a `MermaidChart<PlatformName>` that handles the platform-specific details.

Here, this authenticationProvider is also specific to VSCode and might not be needed in GoogleDocs.